src/components/table/TableCellExpander.tsx

```diff
@@ -0,0 +1,54 @@
+import { CellProps } from 'react-table';
+import { ReactComponent as IconChevronDown } from 'assets/icons/chevronDown.svg';
+import { PropsWithChildren } from 'react';
+
+interface Props {
+  getExpandedContent?: () => JSX.Element;
+  getCollapsedContent?: () => JSX.Element;
+  getCannotExpandContent?: () => JSX.Element;
+  cellData: PropsWithChildren<CellProps<any>>;
+  canExpandMultiple?: boolean;
+}
+export const TableCellExpander = ({
+  cellData,
+  canExpandMultiple = false,
+  getExpandedContent,
+  getCollapsedContent,
+  getCannotExpandContent,
+}: Props) => {
+  const {
+    row: { canExpand, isExpanded, toggleRowExpanded },
+    toggleAllRowsExpanded,
+  } = cellData;
+
+  const handleClick = () => {
+    if (!canExpandMultiple && !isExpanded) {
+      toggleAllRowsExpanded(false);
+    }
+    toggleRowExpanded(!isExpanded);
+  };
+
+  return canExpand ? (
+    <span onClick={() => handleClick()}>
+      {isExpanded ? (
+        getExpandedContent ? (
+          getExpandedContent()
+        ) : (
+          <button className="btn-outline-primary btn-sm rounded-[12px] !w-[35px] !h-[35px] p-0 border shadow-header">
```

Comment:
I suggest you define these fallback render functions outside of the component main function (but here, in this file) and use them as fallback functions when no function is passed.
for example:
const getDefaultExpandedContent = () => (<button> .....);

and then:
export const TableCellExpander = ({
  cellData,
  canExpandMultiple = false,
  getExpandedContent = getDefaultExpandedContent,
  getCollapsedContent = getDefaultCollapsedContent,
  getCannotExpandContent = getDefaultCannotExpandContent,

Reply:
done